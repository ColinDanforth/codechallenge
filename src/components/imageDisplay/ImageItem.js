import React from 'react'
import PropTypes from 'prop-types'

const imageDiv = {
  display: 'flex',
  justifyContent: 'center',
  width: '80%',
  height: 'auto',
  marginBottom: '1%',
  backgroundColor: '#000',
}

const imageHeight = {
  maxHeight: '280px',
  width: 'auto',
  height: 'auto'
}

class ImageItem extends React.Component{
  render(){
    return(
      <div style={imageDiv}>
        <img style={imageHeight} src={this.props.thisItem.images[0].url} alt="test"/>
      </div>
    )
  }
}

export default ImageItem

ImageItem.propTypes={
  thisItem: PropTypes.object
}
