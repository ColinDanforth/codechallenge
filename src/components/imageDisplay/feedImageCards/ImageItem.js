import React from 'react'
import PropTypes from 'prop-types'
import UserDataCard from "../../imageCommon/UserDataCard"
import Title from "../../imageCommon/Title"

const rootCard = {
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  maxWidth: '45vw',
  width: 'auto',
  height: 'auto',
  marginBottom: '1%',
  backgroundColor: '#FFF',
}

const titleDiv = {
  maxWidth: '45vw',
  width: 'auto',
  marginTop: '2%',
}

const dataCardPosition = {
  maxHeight:'200px',
  minHeight: '100px',
  minWidth: '96%',
  width: 'auto',
  height:'auto',
  margin: '2%',
  flex: "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
}

const imageWrapper = {
  flex: '1',
  marginTop: '2%',
  marginBottom: '2%',
}

const imageObjectStyle = {
  maxWidth: '45vw',
  width: 'auto',
  height: 'auto',
  cursor: 'pointer',
}

class ImageItem extends React.Component{
  constructor(){
    super()

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick(){
    this.props.makeFullScreen(this.props.thisItem.id, this.props.thisItem)
  }

  render(){
    return(
      <div style={rootCard}>
        {this.props.thisItem.name.length > 0 ?
          (
            <div style={titleDiv}>
              <Title
                name={this.props.thisItem.name}
                description=''
              />
            </div>
          )
            :
          (<div/>)
        }
        <div style={imageWrapper}>
          <img style={imageObjectStyle} src={this.props.thisItem.image_url} alt="test" onClick={this.handleClick}/>
        </div>
        <div style={dataCardPosition}>
          <UserDataCard dataObject={this.props.thisItem} showMoreInfo={false}/>
        </div>
      </div>
    )
  }
}

export default ImageItem

ImageItem.propTypes={
  thisItem: PropTypes.object,
  makeFullScreen: PropTypes.func,
}
