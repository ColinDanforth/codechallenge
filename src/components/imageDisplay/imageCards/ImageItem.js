import React from 'react'
import PropTypes from 'prop-types'
import DataCard from "./DataCard"
import {Redirect} from 'react-router-dom'

const rootCard = {
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  width: '70%',
  height: 'auto',
  marginBottom: '1%',
  backgroundColor: '#FFF',
}

const imageWrapper = {
  marginTop: '2%',
  marginBottom: '2%',
}

const imageObjectStyle = {
  maxWidth: '840px',
  minWidth: '840px',
  width: 'auto',
  height: 'auto',
  cursor: 'pointer',
}

class ImageItem extends React.Component{
  constructor(){
    super()

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick(){
    this.props.makeFullScreen(this.props.thisItem.id)
  }

  render(){
    return(
      <div style={rootCard}>
        {this.props.thisItem.name.length > 0 ?
          (<h1>{this.props.thisItem.name}</h1>)
            :
          (<div/>)
        }
        <div style={imageWrapper}>
          <img style={imageObjectStyle} src={this.props.thisItem.image_url} alt="test" onClick={this.handleClick}/>
        </div>
        <DataCard dataObject={this.props.thisItem}/>
      </div>
    )
  }
}

export default ImageItem

ImageItem.propTypes={
  thisItem: PropTypes.object,
  makeFullScreen: PropTypes.func,
}
