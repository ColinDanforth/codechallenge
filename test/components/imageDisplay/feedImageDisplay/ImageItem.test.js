import React from 'react';
import ImageItem from '../../../../src/components/imageDisplay/feedImageCards/ImageItem';
import renderer from 'react-test-renderer';

test('ImageItem should display the base ImageItem component with all loaded test data', () => {
  const testData={
    user: {
      avatars: {
        default: {
          https: 'http://google.ca'
        }
      },
      fullname: 'Test User',

    },
    rating: '99.99999995',
    image_url: 'http://google.com',
    name: 'Test name',
    description: 'test description',
  }

  const makeFullScreen = () => {
    return null
  }

  const component = renderer.create(
    <ImageItem key='1' thisItem={testData} makeFullScreen={makeFullScreen}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
