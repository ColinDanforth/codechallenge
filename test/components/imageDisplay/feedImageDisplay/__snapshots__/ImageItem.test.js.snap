// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageItem should display the base ImageItem component with all loaded test data 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "display": "flex",
      "flexDirection": "column",
      "height": "auto",
      "justifyContent": "center",
      "marginBottom": "1%",
      "maxWidth": "45vw",
      "width": "auto",
    }
  }
>
  <div
    style={
      Object {
        "marginTop": "2%",
        "maxWidth": "45vw",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderBottom": "inset",
          "display": "flex",
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "flexDirection": "Column",
          "height": "auto",
          "justifyContent": "center",
          "minWidth": "40vw",
          "width": "auto",
        }
      }
    >
      <h1>
        Test name
      </h1>
      <div
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "flex": "1",
        "marginBottom": "2%",
        "marginTop": "2%",
      }
    }
  >
    <img
      alt="test"
      onClick={[Function]}
      src="http://google.com"
      style={
        Object {
          "cursor": "pointer",
          "height": "auto",
          "maxWidth": "45vw",
          "width": "auto",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "height": "auto",
        "margin": "2%",
        "maxHeight": "200px",
        "minHeight": "100px",
        "minWidth": "96%",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderBottom": "inset",
          "borderRight": "inset",
          "display": "flex",
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "flexDirection": "column",
          "height": "auto",
          "justifyContent": "flex-start",
          "minWidth": "100%",
          "width": "auto",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "flexDirection": "row",
            "height": "auto",
            "justifyContent": "flex-start",
            "minWidth": "100%",
            "width": "auto",
          }
        }
      >
        <img
          alt="avatar"
          src="http://google.ca"
          style={
            Object {
              "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
              "height": "auto",
              "margin": "0.3%",
              "maxHeight": "120px",
              "width": "auto",
            }
          }
        />
        <div
          style={
            Object {
              "display": "flex",
              "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "space-between",
              "marginLeft": "1%",
              "minWidth": "78%",
              "width": "auto",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                "flexDirection": "column",
                "height": "auto",
                "justifyContent": "flex-start",
                "minWidth": "85%",
                "width": "auto",
              }
            }
          >
            <p
              style={
                Object {
                  "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                  "height": "auto",
                  "width": "auto",
                }
              }
            >
              Photographer: 
              Test User
            </p>
            <p
              style={
                Object {
                  "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                  "height": "auto",
                  "width": "auto",
                }
              }
            >
              Rating: 
              99.99999995
              %
            </p>
          </div>
          <div
            style={
              Object {
                "alignSelf": "right",
                "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                "height": "auto",
                "marginTop": "5px",
                "maxHeight": "40px",
                "width": "auto",
              }
            }
          >
            <img
              alt="like"
              src="test-file-stub"
              style={
                Object {
                  "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                  "height": "auto",
                  "maxWidth": "40px",
                  "width": "auto",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
